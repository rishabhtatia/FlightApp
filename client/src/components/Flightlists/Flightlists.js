import React from 'react';
import _ from 'lodash';
import Flightlist from './Flightlist/Flighlist';
import MultipleFlights from './Multipleflights/Multipleflights';
import styles from './Flightlists.module.css';

const FlightLists = ({ data }) => {
  return (
    <div className={styles.flightLists}>
      <h1>FLIGHT LISTS</h1>
      {_.map(data, (item) =>
        Array.isArray(item) ? (
          <MultipleFlights
            data={item}
            key={item[0].flightNo + item[item.length - 1].flightNo}
          />
        ) : (
          <Flightlist data={item} key={item.flightNo} />
        )
      )}
    </div>
  );
};

export default FlightLists;
