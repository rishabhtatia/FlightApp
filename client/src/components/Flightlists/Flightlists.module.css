.flightLists {
  margin: 0 3px;
  border: 1px solid black;
}
