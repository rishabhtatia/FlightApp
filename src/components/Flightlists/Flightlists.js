import React from 'react';
import _ from 'lodash';
import FLIGHTLIST from './Flightlist/Flighlist';
import MULTIPLEFLIGHTS from './Multipleflights/Multipleflights';
import styles from './Flightlists.module.css';

const FLIGHTLISTS = ({ data }) => {
  return (
    <div className={styles.flightLists}>
      <h1>FLIGHT LISTS</h1>
      {_.map(data, (item) =>
        Array.isArray(item) ? (
          <MULTIPLEFLIGHTS
            data={item}
            key={item[0].flightNo + item[item.length - 1].flightNo}
          />
        ) : (
          <FLIGHTLIST data={item} key={item.flightNo} />
        )
      )}
    </div>
  );
};

export default FLIGHTLISTS;
