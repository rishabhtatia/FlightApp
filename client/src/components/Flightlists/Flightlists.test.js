import React from 'react';
import renderer from 'react-test-renderer';

import Flightlists from './Flightlists';

it('renders when there is one item each for single and multi flights', () => {
  const item = [
    {
      arrivalTime: '10:20',
      date: '2020/11/01',
      departureTime: '8:10',
      destination: 'Delhi (DEL)',
      flightNo: 'AI-104',
      name: 'Air India',
      origin: 'Pune (PNQ)',
      price: 4681,
    },
    [
      {
        arrivalTime: '7:10',
        date: '2020/11/01',
        departureTime: '6:10',
        destination: 'Mumbai (BOM)',
        flightNo: 'AI-103',
        name: 'Air India',
        origin: 'Pune (PNQ)',
        price: 2537,
      },
      {
        arrivalTime: '11:30',
        date: '2020/11/01',
        departureTime: '9:00',
        destination: 'Delhi (DEL)',
        flightNo: 'AI-110',
        name: 'Air India',
        origin: 'Mumbai (BOM)',
        price: 4137,
      },
    ],
  ];
  const tree = renderer.create(<Flightlists data={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when there is one item for single flight  and zero for multi flights', () => {
  const item = [
    {
      arrivalTime: '10:20',
      date: '2020/11/01',
      departureTime: '8:10',
      destination: 'Delhi (DEL)',
      flightNo: 'AI-104',
      name: 'Air India',
      origin: 'Pune (PNQ)',
      price: 4681,
    },
    [],
  ];
  const tree = renderer.create(<Flightlists data={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});
