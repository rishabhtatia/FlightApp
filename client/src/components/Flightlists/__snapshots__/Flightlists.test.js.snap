// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is one item each for single and multi flights 1`] = `
<div
  className="flightLists"
>
  <h1>
    FLIGHT LISTS
  </h1>
  <div
    className="flightList"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="row"
        >
          <div
            className="col-sm"
          >
            <h4>
              Air India
            </h4>
            <span>
              AI-104
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4>
              8:10
            </h4>
            <span>
              Pune (PNQ)
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4>
              10:20
            </h4>
            <span>
              Delhi (DEL)
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4>
              2h:10m
            </h4>
            <span>
              Non Stop
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4
              className="redColor"
            >
              ₹4,681.00
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="multipleFlights"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="row"
        >
          <div
            className="col-sm"
          >
            <h4>
              Multiple
            </h4>
            <button
              className="expandFlagButton"
              onClick={[Function]}
            >
              ShowDetails
            </button>
          </div>
          <div
            className="col-sm"
          >
            <h4>
              6:10
            </h4>
            <span>
              Pune (PNQ)
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4>
              11:30
            </h4>
            <span>
              Delhi (DEL)
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4
              className="totalDuration"
            >
              5h:20m
            </h4>
            <span>
              Total Duration
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4
              className="redColor"
            >
              ₹6,674.00
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders when there is one item for single flight  and zero for multi flights 1`] = `
<div
  className="flightLists"
>
  <h1>
    FLIGHT LISTS
  </h1>
  <div
    className="flightList"
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="row"
        >
          <div
            className="col-sm"
          >
            <h4>
              Air India
            </h4>
            <span>
              AI-104
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4>
              8:10
            </h4>
            <span>
              Pune (PNQ)
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4>
              10:20
            </h4>
            <span>
              Delhi (DEL)
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4>
              2h:10m
            </h4>
            <span>
              Non Stop
            </span>
          </div>
          <div
            className="col-sm"
          >
            <h4
              className="redColor"
            >
              ₹4,681.00
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
